irods/manager/user_manager.py

```diff
@@ -89,10 +89,12 @@ def create(self, user_name, user_type, user_zone="", auth_str=""):
         logger.debug(response.int_info)
         return self.get(user_name, user_zone)
 
-    def remove(self, user_name, user_zone=""):
+    def remove(self, user_name, user_zone="", _object = None):
+        if _object is None:
+            _object = self.get(user_name, user_zone)
         message_body = GeneralAdminRequest(
             "rm",
-            "user",
+            "user" if (_object.type != "rodsgroup" or self.sess.server_version < (4, 3, 2)) else "group",
```

Comment:
It seems that there is a codepath in which `type` may not be defined for `_object`: when `result == None` in `__init__` for `iRODSUser`. This seems to be an exceptional case, though, so maybe this is fine?

Reply:
In this case, probably not a concern. This line protects us from `result` being `None`:
https://github.com/irods/python-irodsclient/blob/ed2e73cb79dcf08efedeed4a19f3e2a43db90f9c/irods/manager/user_manager.py#L51

Reply:
Excellent. Resolving.